collections/mod.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
```

Comment:
What's the benefit of these over `Array<T>` ?

Reply:
It restricts the API to amortized O(1) ops.

Reply:
Hmm. That should be explained in the jsdoc.

I think you could combine the two into one class and still satisfy the API goal of O(1) ops.

Reply:
Stack and Queue are distinct data-structures, and useful to have.

I see what you're saying about combining, since the Queue data-structure could trivially have O(1) (right) push, peek, and pop. ~I can't recall what that datastructure is called.~ Oh yeah, that's a Deque, I should add that as well.

Sometimes you want a Queue... and you want to enforce items not jumping the queue.

Reply:
Maybe "trivially" isn't the right word: I'm not sure how to do Deque.

(Without a linked list.)